_data/engines.yml

```diff
@@ -1,4 +1,236 @@
 ---
+- name: AISA
+  url: http://www.aisa.digital/machine-translation-api
+  languages: [ ar, de, es-es, fr-fr, hi, huyu, id, it, ja, ko, ms, pt-pt, ru, th, vi, zh-cn, zh-tw ]
+
+- name: Alexa AI
+  url: http://alexatranslations.com/alexa-translations-a-i
+  languages: [ en, es-419, fr-ca, pt-br, zh ]
+
+- name: Alibaba Translate
+  url: http://translate.alibaba.com
+  languages: [ ar, de, en, es, fr, id, it, ja, ko, pl, pt, ru, th, tr, vi, zh, [zh-cn, zh-tw], [zh-tw, zh-cn], [zh-cn, yue] ]
+  info_url: https://www.alibabacloud.com/help/en/machine-translation/latest/general-version-of-machine-translation
+
+- name: Amazon Translate
+  url: http://aws.amazon.com/es/translate
+  languages: [ af, am, ar, az, bg, bn, bs, ca, cs, cy, da, de, el, en, es, es-mx, et, fa, fa-af, fi, fr, fr-ca, ga, gu, ha, he, hi, hr, ht, hu, hy, id, is, it, ja, ka, kk, kn, ko, lt, lv, mk, ml, mn, mr, ms, mt, nl, no, pa, pl, ps, pt, pt-pt, ro, ru, si, sk, sl, so, sq, sr, sv, sw, ta, te, th, tl, tr, uk, ur, uz, vi, zh, zh-tw ]
+  info_url: https://docs.aws.amazon.com/translate/latest/dg/what-is-languages.html
+
+- name: Apertium
+  url: http://www.apertium.org
+  languages: [ afr-nld, arg-cat, bel-rus, br-fr, cat-ita, cat-srd, crh-tur, cym-eng, dan-nor, eng-cat, eng-kaz, en-gl, eng-spa, eo-ca, eo-en, eo-es, eo-fr, es-gl, es-pt, es-ro, eu-en, eu-es, fra-cat, fr-es, hbs-eng, hbs-mkd, hbs-slv, ind-zlm, isl-eng, isl-swe, kaz-rus, kaz-tat, mkd-bul, mkd-eng, mlt-ara, nno-nob, oc-ca, oc-es, oci-fra, pol-szl, por-cat, pt-gl, ron-cat, rus-ukr, sme-nob, spa-arg, spa-ast, spa-cat, spa-ita, srd-ita, swe-dan, swe-nor, urd-hin ]
```

Comment:
Need to fix this

```
[ [ af, nl ] ...
```

Reply:
Fixed!
I will add 3-letter codes to languages.yml too, especially for Niutrans supported languages.